File: Introduction-To-Algorithm-Fourth-Edition/04-Divide-and-Conquer/4-1-Multiplying-square-matrices/Code-Matrix-Multiplication/Matrix-Multiplication.py
def matrix_multiply(A,B):
    n=len(A)
    m=len(A[0])
    p=len(B[0])
    if m!=len(B):
        print("Matrix dimension are not compatible for multiplication.")
        return []
    C=[[0 for _ in range(p)] for _ in range(n)]
    for i in range(n):
        for j in range(p):
            for k in range(m):
                C[i][j]+=A[i][k]*B[k][j]
    return C
A=[[1,2,3],[4,5,6]]
B=[[7,8],[9,10],[11,12]]
result=matrix_multiply(A,B)
for row in result:
    print(row)

File: Introduction-To-Algorithm-Fourth-Edition/04-Divide-and-Conquer/4-1-Multiplying-square-matrices/Code-Matrix-Multiplication/Matrix-Multiplication.cpp
#include<iostream>
#include<vector>
using namespace std;
vector<vector<int>> matrixMultiply(const vector<vector<int>>& A,const vector<vector<int>>& B)
{
    int n=A.size();
    int m=A[0].size();
    int p=B[0].size();
    if(m!=B.size())
    {
        cout<<"Matrix dimension are not comatible for multiplication"<<endl;
        return {};
    }
    vector<vector<int>> C(n,vector<int> (p,0));
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<p;j++)
        {
            for(int k=0;k<m;k++)
            {
                C[i][j]+=A[i][k]*B[k][j];
            }
        }
    }
    return C;
}
int main()
{
    vector<vector<int>> A={{1,2,3},{4,5,6}};
    vector<vector<int>> B={{7,8},{9,10},{11,12}};
    vector<vector<int>> result=matrixMultiply(A,B);
    for(int i=0;i<result.size();i++)
    {
        for(int j=0;j<result[i].size();j++)
        {
            cout<<result[i][j]<<" ";
        }
        cout<<endl;
    }
    return 0;
}